### PR TITLE
Add Prefixed Rector

### LINKED ISSUE #177: Create PHAR with prefixes to avoid class and package version conflicts
Is a `.phar` build planned for this tool? I would be interested. :-)

### PR DESCRIPTION
**Pull request by TomasVotruba**
Follow up to #449 

Closes #177

### Usage

```bash
bin/build-prefixed-rector.sh
bin/publish-prefixed-rector.sh
```

### Is Published on

https://github.com/rectorphp/rector-prefixed

### Todobin/publish-prefixed-rector.sh

- [x] add bash to distribute generated code to `rector/prefixed-rector` package - https://github.com/rectorphp/rector-prefixed

- [x] make `composer.json` work as expected, drop require sections probably 
- [x] test install `composer require rector/rector-prefixed` in project with phpstan 0.9 and php-parser 3.0
- [x] test run `rector/rector-prefixed` in symfony/demo project

    ```bash
    composer create-project symfony/symfony-demo
    composer require rector/rector-prefixed:@dev --dev

    # passes
    vendor/bin/rector
    vendor/bin/rector process src/Controller/BlogController.php --level action-injection-to-constructor-injection -vvv
    ```
     

- [x] update `README.md`

### Waits on

- [x] https://github.com/humbug/php-scoper/pull/203, nice to have

- [x] https://github.com/humbug/php-scoper/issues/210

### CODE REVIEW

**Comment by TomasVotruba** [comment]
- [ ] add bash to distribute generated code to `rector/prefixed-rector` package

File: src/Autoloading/AdditionalAutoloader.php [hunk 1]
```diff
@@ -72,7 +72,7 @@ private function autoloadDirectoriesFromParameter(ParameterProvider $parameterPr
     private function autoloadDirectories(array $directories): void
     {
         $robotLoader = new RobotLoader();
-        $robotLoader->ignoreDirs = ['*Fixtures'] + $robotLoader->ignoreDirs;
+        $robotLoader->ignoreDirs[] = '*Fixtures';
```
Comment (TomasVotruba) [inline src/Autoloading/AdditionalAutoloader.php]: nette/robot-loader 3.0.3 BC break

**Comment by mssimi** [comment]
After super quick review [here](https://github.com/rectorphp/rector-prefixed/blob/6b690e46e54830a944618d3a2bf50a7c2bd13939/src/Bridge/Symfony/NodeAnalyzer/ControllerMethodAnalyzer.php#L16) is example of problem, which may be vast.

**Comment by TomasVotruba** [comment]
> Maybe update our installation guide?

Sure, I added it to @todo of the PR

**Comment by TomasVotruba** [comment]
Looks like bug in `php-scoper`.

```php
# scoper.inc.php

<?php declare(strict_types=1);

require_once __DIR__ . '/vendor/autoload.php';

use Isolated\Symfony\Component\Finder\Finder;

return [
    'prefix' => 'RectorPrefixed',
    'finders' => [
        Finder::create()
            ->files()
            ->in(__DIR__ .'/test-me')
    ]
];
```

Turns this: https://github.com/phpDocumentor/ReflectionDocBlock/blob/master/src/DocBlock/StandardTagFactory.php

to 

```php
   private $tagHandlerMappings = [
        'author' => 'RectorPrefixed\\phpDocumentor\\Reflection\\DocBlock\\Tags\\Author',
        'covers' => 'RectorPrefixed\\phpDocumentor\\Reflection\\DocBlock\\Tags\\Covers',
        'deprecated' => 'RectorPrefixed\\phpDocumentor\\Reflection\\DocBlock\\Tags\\Deprecated',
        // 'example'        => '\phpDocumentor\Reflection\DocBlock\Tags\Example',
        'link' => 'RectorPrefixed\\phpDocumentor\\Reflection\\DocBlock\\Tags\\Link',
        'method' => 'RectorPrefixed\\phpDocumentor\\Reflection\\DocBlock\\Tags\\Method',
        'param' => 'RectorPrefixed\\phpDocumentor\\Reflection\\DocBlock\\Tags\\Param',
        'property-read' => 'RectorPrefixed\\phpDocumentor\\Reflection\\DocBlock\\Tags\\PropertyRead',
        'property' => 'RectorPrefixed\\phpDocumentor\\Reflection\\DocBlock\\Tags\\Property',
        'property-write' => 'RectorPrefixed\\phpDocumentor\\Reflection\\DocBlock\\Tags\\PropertyWrite',
        'return' => '\\phpDocumentor\\Reflection\\DocBlock\\Tags\\Return_',
        'see' => 'RectorPrefixed\\phpDocumentor\\Reflection\\DocBlock\\Tags\\See',
        'since' => 'RectorPrefixed\\phpDocumentor\\Reflection\\DocBlock\\Tags\\Since',
        'source' => 'RectorPrefixed\\phpDocumentor\\Reflection\\DocBlock\\Tags\\Source',
        'throw' => 'RectorPrefixed\\phpDocumentor\\Reflection\\DocBlock\\Tags\\Throws',
        'throws' => 'RectorPrefixed\\phpDocumentor\\Reflection\\DocBlock\\Tags\\Throws',
        'uses' => 'RectorPrefixed\\phpDocumentor\\Reflection\\DocBlock\\Tags\\Uses',
        'var' => '\\phpDocumentor\\Reflection\\DocBlock\\Tags\\Var_',
        'version' => 'RectorPrefixed\\phpDocumentor\\Reflection\\DocBlock\\Tags\\Version',
    ];
```

**Comment by TomasVotruba** [comment]
Waits on https://github.com/humbug/php-scoper/issues/210

**Comment by TomasVotruba** [comment]
It works: :+1:

**Comment by TomasVotruba** [comment]
Huge thanks to @mssimi who pushed this a lot further than I could every imagine.
It was much easier to finish this with your help. I would not make it without you :handshake: 

I believe few bugs will appear, but basic workflow to generate prefixed version is done. I'm very happy with this result :parasol_on_ground:

**Pull request merged**